src/Form/ParagraphsPasteForm.php

```diff
@@ -292,10 +292,32 @@ public static function getThirdPartyForm(WidgetInterface $plugin, $field_name) {
         'visible' => [":input[name=\"fields[$field_name][settings_edit_form][third_party_settings][paragraphs_paste][split_method][regex]\"]" => ['checked' => TRUE]],
         'required' => [":input[name=\"fields[$field_name][settings_edit_form][third_party_settings][paragraphs_paste][split_method][regex]\"]" => ['checked' => TRUE]],
       ],
-
+      '#element_validate' => [[__CLASS__, 'validateRegEx']],
     ];
 
     return $elements;
   }
 
+  /**
+   * Validates the regex field element.
+   *
+   * @param array $element
+   *   The form element to process.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $complete_form
+   *   The complete form structure.
+   */
+  public static function validateRegEx(array &$element, FormStateInterface $form_state, array &$complete_form) {
+
+    $parents = $element['#parents'];
+    array_pop($parents);
+    $parents[] = 'split_method';
+
+    $split_methods = $form_state->getValue($parents);
+    if (empty($form_state->getValue($element['#parents'])) && !empty($split_methods['regex'])) {
```

Comment:
I would suggest naming `$parents` something else, it gets confusing having that intermixed with `$element['#parents']`.